system/lib/wasmfs/js_api.cpp

```diff
@@ -190,8 +202,11 @@ void _wasmfs_readdir_finish(struct wasmfs_readdir_state* state) {
 
 char* _wasmfs_get_cwd(void) {
   // TODO: PATH_MAX is 4K atm, so it might be good to reduce this somehow.
-  static thread_local char path[PATH_MAX];
-  return getcwd(path, PATH_MAX);
+  static thread_local std::unique_ptr<std::vector<char>> path;
```

Comment:
And here I think you can do `static thread_local std::vector<char> path`;  followed by `path->resize(PATH_MAX)`

Reply:
Actually I think I might prefer the simple C version that just looks like:

```
static thread_local char* buffer;
buffer = realloc(buffer, <SIZE>);
```

Even though its less C++'y I think it more obvious / clear / concise.

Reply:
Oh I see, are you trying to free the memory on shutdown.. that seems rather unnecessary though.